src/Storages/MergeTree/MergeTreeData.cpp

```diff
@@ -1239,7 +1240,7 @@ void MergeTreeData::MergingParams::check(const MergeTreeSettings & settings, con
     if (mode == MergingParams::Collapsing)
         check_sign_column(false, "CollapsingMergeTree");
 
-    if (mode == MergingParams::Summing)
+    if (mode == MergingParams::Summing || mode == MergingParams::Coalescing)
```

Comment:
After adding the new condition, both Summing and Coalescing modes share the same `allow_summing_columns_in_partition_or_order_key` setting.
This is slightly misleading.

There should either be a separate setting for the coalescing transform, or the existing setting should be renamed to `allow_summing_and_coalescing_columns_in_partition_or_order_key`, with `allow_summing_columns_in_partition_or_order_key` retained as an alias for compatibility.

Please add a test for this setting.

Reply:
I added new setting, but I think tests is useless, because this setting just indicates either we should throw exception or not